bliss/models/encoder.py

```diff
@@ -228,11 +225,13 @@ def __init__(
         self.log_softmax = nn.LogSoftmax(dim=1)
 
         # get index for prob_n_sources, assigned indices that were not used.
-        self.indx_mats, last_indx = self._get_hidden_indices()
-        self.prob_n_source_indx = torch.arange(
-            last_indx,
-            self.dim_out_all,
-            device=device,
+        indx_mats, last_indx = self._get_hidden_indices()
+        for k, v in indx_mats.items():
+            self.register_buffer(k, v, persistent=False)
```

Comment:
now that the matrices are an attribute I guess I would rename them something like `k+'_indx'` to avoid confusion like the one below

Reply:
This was so that attribute names align with `self.variational_params` keys.
https://github.com/prob-ml/bliss/blob/1802f0aefdc44d0bb482f1575f328954b4ddfe4c/bliss/models/encoder.py#L362-L363
But I can make those changes.